src/Fmt_ast.ml

```diff
@@ -1880,14 +1880,29 @@ and fmt_expression c ?(box = true) ?pro ?epi ?eol ?parens ?(indent_wrap = 0)
             (sub_mod ~ctx body_me, Some (sub_mty ~ctx body_mt))
         | _ -> (xbody, None)
       in
+      let let_module =
+        match c.conf.let_module with
+        | `Compact ->
+            fmt_module c keyword name xargs (Some xbody) true xmty []
+              ~epi:(fmt "in")
+        | `Sparse ->
+            hovbox 2
```

Comment:
Would it be possible to implement `compact` by sharing the code with `sparse` ? (eg. using the `Params` architecture)
It makes the code a little bit hard to read. Also the `xargs, xbody` code is only used by the `compact` branch.

Reply:
I tried to do that, but the box layout is very different in the 2 cases, but maybe it can be done later as more of the formatting data is moved to `Params` anyway.